library/std/src/io/copy.rs

```diff
@@ -249,7 +249,6 @@ impl<I: Write + ?Sized> BufferedWriterSpec for BufWriter<I> {
                 }
             } else {
                 self.flush_buf()?;
```

Comment:
`flush_buf` doesn't deinitialize the filled part of the buffer and will have written the complete buffer if it returns. As the previously filled portion will now be returned as part of `spare_capacity_mut`, I think we can add the old buffer length to `init`.